### PR TITLE
Load sets

### PR DESCRIPTION
**Pull request by roppazvan**
* Load tokens when adding the external set node to the stage

https://github.com/tokens-studio/graph-engine/assets/21245398/3e4411c0-5648-4cc2-9738-cbb1a3aae99c

### CODE REVIEW

File: packages/graph-editor/src/components/flow/nodes/sets/ExternalSetNode.tsx [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+import { useExternalData } from '#/context/ExternalDataContext.tsx';
+
+
+type ExternalSetData = {
+    urn: string
+    name: string;
+};
+
+const ExternalSetNode: FC<NodeProps<ExternalSetData>> = () => {
+    const { setControls, setTitle, state, output, setState } = useNode();
```
Comment (SorsOps) [inline packages/graph-editor/src/components/flow/nodes/sets/ExternalSetNode.tsx:26]: We need to expose the ephemeral property through useNode and the rest of the ephemeral lifecycle. That would remove the complexity of each node managing its lifecycle to load external data

File: packages/graph-engine/src/nodes/set/externalTokens.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import { NodeDefinition, NodeTypes } from "../../types.js";
+import { SingleToken } from "@tokens-studio/types";
+
+export const type = NodeTypes.SET;
```
[the remark this answers is not in the document]
Comment (roppazvan) [reply packages/graph-engine/src/nodes/set/externalTokens.ts:4]: but we have a different type INLINE_SET

File: packages/graph-editor/src/index.tsx [hunk 3]
```diff
@@ -1,5 +1,7 @@
-export * from './utils/index.ts';
-export * from './types/index.ts';
-export * from './redux/index.tsx';
-export * from './editor/index.tsx';
+export * from './components';
```
Comment (SorsOps) [inline packages/graph-editor/src/index.tsx:1]: Are these working without throwing an warnings/errors? The resolution type we were using before  necessitated that we used full paths

File: packages/graph-editor/package.json [hunk 4]
```diff
@@ -72,7 +72,8 @@
   },
   "peerDependencies": {
     "react": "18.2.0",
-    "react-dom": "18.2.0"
+    "react-dom": "18.2.0",
+    "@tokens-studio/sdk": "^0.3.0"
```
Comment (SorsOps) [inline packages/graph-editor/package.json:76]: Looks like we can remove this as its only used in the graph engine ?

**Review by SorsOps — CHANGES_REQUESTED** [review]
Looks really good otherwise

File: packages/graph-editor/src/index.tsx (hunk 3 above)
Comment (roppazvan) [reply packages/graph-editor/src/index.tsx:1]: yes, working

File: packages/graph-editor/package.json (hunk 4 above)
Comment (roppazvan) [reply packages/graph-editor/package.json:76]: It's actually used in  ExternalDataContext.tsx

**Review by SorsOps — APPROVED** [review]

**Pull request merged**